lib/docker/imageservice.go

```diff
@@ -336,29 +382,59 @@ func initTransport(req RegistryConnectionRequest) (*http.Transport, string) {
 		DisableKeepAlives:   true,
 	}
 
-	cert, err := tls.LoadX509KeyPair(req.ClientCertPath, req.ClientKeyPath)
-	if err == nil {
-		roots, err := newCertPool([]string{req.CACertPath})
-		if err == nil {
+	// Figure out the registry address scheme (http or https). If the scheme
+	// was specified explicitly, keep it as-is. Otherwise, default to https
+	// unless the insecure flag was provided.
+	defaultScheme := "https"
+	if req.Insecure {
+		defaultScheme = "http"
+	}
+	registryAddress := utils.EnsureScheme(req.RegistryAddress, defaultScheme)
+
+	// If the scheme was set explicitly to https, along with the insecure
+	// flag, ignore the certificate error (this is what Docker does too).
+	var err error
+	if req.Insecure && strings.HasPrefix(registryAddress, "https://") {
+		transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	} else if strings.HasPrefix(registryAddress, "https://") {
+		transport.TLSClientConfig, err = req.TLSClientConfig()
+		if err != nil {
+			log.WithError(err).Debugf("No TLS trust for %s.", req)
```

Comment:
It seems a bit odd to log the error and drop it at Debug level if the trust store cannot be initialized. My preferred approach would be to test if the trust store config has been passed (CACertPath, etc) and hard error if that configuration cannot be used.

Reply:
Yeah, this is a side effect of the preexisting behavior unfortunately - these paths are actually always being passed and set to the default locations inside Gravity clusters so I kept this for now for backwards compatibility. I've updated this to use a more secure TLS config though in case system trust store is used.